### PR TITLE
Hb4

### PR DESCRIPTION
**Pull request by jah701**

### CODE REVIEW

File: src/main/java/com/dev/cinema/security/AuthenticationService.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.dev.cinema.security;
+
+import com.dev.cinema.exceptions.AuthenticationException;
+import com.dev.cinema.model.User;
+
+public interface AuthenticationService {
+    User login(String email, String password) throws AuthenticationException;
+
+    /**
+     * We should register a new user. New user entity will contains the email and password
```
Comment (Rudiachenko) [inline src/main/java/com/dev/cinema/security/AuthenticationService.java]: do we need this comment here?

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    @Override
+    public User register(String email, String password) {
+        User user = new User(password, email);
+        user = userService.add(user);
+        shoppingCartService.registerNewShoppingCart(user);
+        return user;
+
+    }
+
+    private boolean isValid(String password, User user) {
```
Comment (Rudiachenko) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java]:
```suggestion
    private boolean isPasswordValid(String password, User user) {
```

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.ShoppingCartService;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
```
Comment (Rudiachenko) [inline src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]: do we need this separate?

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
+        shoppingCart.setTickets(List.of(ticket));
```
Comment (Rudiachenko) [inline src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]: here we should add one else ticket but not change the old value

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java (hunk 3 above)
Comment (jah701) [reply src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java:19]: For me its more comfortable to have one more line between injectors :)

File: src/main/java/com/dev/cinema/security/AuthenticationService.java (hunk 1 above)
Comment (jah701) [reply src/main/java/com/dev/cinema/security/AuthenticationService.java]: fixed

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java (hunk 4 above)
Comment (jah701) [reply src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]: Agree. Fixed

File: src/main/java/com/dev/cinema/dao/UserDao.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.dev.cinema.dao;
+
+import com.dev.cinema.model.User;
+import java.util.Optional;
+
+public interface UserDao {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/UserDao.java:6]: this should have been in previous PR

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        }
+        throw new AuthenticationException("Incorrect data entered");
+    }
+
+    @Override
+    public User register(String email, String password) {
+        User user = new User(password, email);
+        user = userService.add(user);
+        shoppingCartService.registerNewShoppingCart(user);
+        return user;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java:34]: in this PR we should only have changes in this method but not the whole class

**Review by dmytro-hryhoruk — COMMENTED** [review]
Refactor your PR so there are only those classes which a related to this HW

**Pull request closed without merging**